File: code/java/flatten.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public void flatten(TreeNode root) {
        if(root == null || (root.left==null && root.right == null) )
            return;
        List<TreeNode> temp = new ArrayList<TreeNode>();
        store(root,temp);
        root.left = null;
        for(int i=1;i<temp.size();i++){
            root.left = null;
            root.right = temp.get(i);
            root = root.right;
        }
    }

    
    public void store(TreeNode root,List<TreeNode> temp){
        if(root == null)
            return;
        temp.add(root); 
        store(root.left,temp);
        store(root.right,temp);
    }
}

File: code/python/flatten.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def flatten(self, root: TreeNode) -> None:
        """
        Do not return anything, modify root in-place instead.
        """
        if root == None:
            return
        if root.left == None and root.right == None:
            return 
        temp = []
        self.store(root,temp)
        root.left = None

        for i in range(1,len(temp)):
            root.right = temp[i]
            root.left = None
            root = root.right

    
    def store(self,root:TreeNode,temp) -> None:
        if root == None:
            return
        temp.append(root)  
        self.store(root.left,temp)
        self.store(root.right,temp)
        return
